src/Illuminate/Routing/ResponseFactory.php

```diff
@@ -138,6 +138,24 @@ public function download($file, $name = null, array $headers = [], $disposition
     }
 
     /**
+     * Return the raw contents of a binary file.
+     *
+     * @param  \SplFileInfo|string $file
+     * @param  array $headers
+     * @return  \Symfony\Component\HttpFoundation\BinaryFileResponse
+     */
+    public function file($file, array $headers = [])
+    {
+        $response = new BinaryFileResponse($file, 200, $headers, true);
+
+		$mime = $response->getFile()->getMimeType();
+
+		$headers['Content-Type'] = $mime;
```

Comment:
The `$headers` have already been sent to the response object.

Use this instead:

```php
$response->headers->set('Content-Type', $mime);
```